numba/npyufunc/array_exprs.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import print_function, division, absolute_import
+import ast
+
+from numpy import ufunc
+
+from .. import ir, types, rewrites, six, config
+from ..typing import npydecl
+from ..targets import npyimpl
+from .dufunc import DUFunc
+
+
+@rewrites.register_rewrite
```

Comment:
Since the order of rewrites will probably be significant, I wonder if it's a good idea to register them by decorator. Generated code could easily change without noticing when reordering imports or shuffling code around. What do you think?